app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -0,0 +1,426 @@
+class ManageIQ::Providers::Openstack::Inventory::Parser::CloudManager < ManagerRefresh::Inventory::Parser
+  include ManageIQ::Providers::Openstack::RefreshParserCommon::HelperMethods
+  include ManageIQ::Providers::Openstack::RefreshParserCommon::Images
+
+  def parse
+    availability_zones
+    cloud_services
+    flavors
+    miq_templates
+    key_pairs
+    orchestration_stacks
+    quotas
+    vms
+    cloud_tenants
+    vnfs
+    vnfds
+  end
+
+  def availability_zones
+    collector.availability_zones.each do |az|
+      availability_zone = persister.availability_zones.find_or_build(az.zoneName)
+      availability_zone.type = "ManageIQ::Providers::Openstack::CloudManager::AvailabilityZone"
+      availability_zone.ems_ref = az.zoneName
+      availability_zone.name = az.zoneName
+    end
+    # ensure the null az exists
+    null_az = persister.availability_zones.find_or_build("null_az")
+    null_az.type = "ManageIQ::Providers::Openstack::CloudManager::AvailabilityZoneNull"
+    null_az.ems_ref = "null_az"
+  end
+
+  def cloud_services
+    related_infra_ems = collector.manager.provider.try(:infra_ems)
+    hosts = related_infra_ems.try(:hosts)
+
+    collector.cloud_services.each do |s|
+      host = hosts.try(:find) { |h| h.hypervisor_hostname == s.host.split('.').first }
+      system_services = host.try(:system_services)
+      system_service = system_services.try(:find) { |ss| ss.name =~ /#{s.binary}/ }
+
+      cloud_service = persister.cloud_services.find_or_build(s.id)
+      cloud_service.ems_ref = s.id
+      cloud_service.source = 'compute'
+      cloud_service.executable_name = s.binary
+      cloud_service.hostname = s.host
+      cloud_service.status = s.state
+      cloud_service.scheduling_disabled = s.status == 'disabled'
+      cloud_service.scheduling_disabled_reason = s.disabled_reason
+      cloud_service.host = host
+      cloud_service.system_service = system_service
+      cloud_service.availability_zone = persister.availability_zones.lazy_find(s.zone)
+    end
+  end
+
+  def cloud_tenants
+    collector.tenants.each do |t|
+      tenant = persister.cloud_tenants.find_or_build(t.id)
+      tenant.type = "ManageIQ::Providers::Openstack::CloudManager::CloudTenant"
+      tenant.name = t.name
+      tenant.description = t.description
+      tenant.enabled = t.enabled
+      # tenant.ems_ref = t.id
```

Comment:
on the other hand, it does not hurt to assign anything. Which is especially handy when we change the manager_ref, e.g. from the :name to :ems_ref